mlflow/langchain/__init__.py

```diff
@@ -608,15 +639,13 @@ def _load_model(
     from langchain.chains.loading import load_chain
     from mlflow.langchain.retriever_chain import _RetrieverChain
 
-    special_chains = _get_map_of_special_chain_class_name_to_kwargs_name()
-
     model = None
-    if key := special_chains.get(model_type):
```

Comment:
This breaks for subclasses of special chains. It also breaks if the chain class name changes in a future version of langchain